projects/can_slave/scripts/read_input.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+"""Test CAN RX <-> UART script.
+"""
+import struct
+import serial
+import serial.tools.list_ports
+
+PACKET_FMT = '3cBIQc'
+
+def select_device():
+    """User-provided serial device selector.
+
+    Args:
+        None
+
+    Returns:
+        The selected serial device as ListPortInfo.
+    """
+    while True:
+        print('Pick the serial device:')
+        ports = serial.tools.list_ports.comports()
+        for i, port in enumerate(ports):
+            print('{}: {}'.format(i, port))
+
+        try:
+            chosen_port = ports[int(input())]
+            print('Selected {}'.format(chosen_port))
+            return chosen_port
+        except IndexError:
+            print('Invalid index!')
+            continue
+
+def parse_line(line):
+    """Parses the input line as an RX packet.
+
+    Args:
+        line: The input line. Should end with a newline.
+
+    Returns:
+        None
+    """
+    if len(line) != 17:
```

Comment:
Maybe move the the size of a `CanUartPacket` (plus the trailing newline character) out as a constant?  `17` isn't really clear.

Reply:
I'll use `struct.calcsize(PACKET_FMT)` instead.